core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -94,9 +98,23 @@ private HTTPClient(
 
     HttpClientBuilder clientBuilder = HttpClients.custom();
 
+    String connectionTimeout = properties.get(REST_CONNECTION_TIMEOUT);
+    String socketTimeout = properties.get(REST_SOCKET_TIMEOUT);
+
+    ConnectionConfig.Builder connConfigBuilder = ConnectionConfig.custom();
+
+    if (connectionTimeout != null) {
+      connConfigBuilder.setConnectTimeout(Long.parseLong(connectionTimeout), TimeUnit.SECONDS);
+    }
+
+    if (socketTimeout != null) {
+      connConfigBuilder.setSocketTimeout(Integer.parseInt(socketTimeout), TimeUnit.SECONDS);
```

Comment:
Do we need to set default values for these two properties.

Reply:
Answered above